tests/data/test_old_get_condition_data.py

```diff
@@ -0,0 +1,490 @@
+        )
+        dm_new = DataManager(
+            adata_perturbation.copy(),
+            sample_rep="X",
+            split_covariates=split_covariates,
+            control_key="control",
+            perturbation_covariates=perturbation_covariates,
+            perturbation_covariate_reps=perturbation_covariate_reps,
+            sample_covariates=["cell_type"],
+            sample_covariate_reps={"cell_type": "cell_type"},
+            primary_group=primary_group,
+        )
+        old = _get_condition_data_old(
+            dm=dm_old,
+            split_cov_combs=dm_old._get_split_cov_combs(adata_perturbation.obs),
+            adata=adata_perturbation.copy(),
+        )
+        new = dm_new._get_condition_data(
+            adata=adata_perturbation.copy(),
+        )
+        compare_train_data(old, new)
+
+
+class TestValidationData:
+    @pytest.mark.parametrize("sample_rep", ["X", "X_pca"])
+    @pytest.mark.parametrize("split_covariates", [[], ["cell_type"]])
+    @pytest.mark.parametrize("perturbation_covariates", perturbation_covariate_comb_args)
+    @pytest.mark.parametrize("perturbation_covariate_reps", [{}, {"drug": "drug"}])
+    def test_get_validation_data(
+        self,
+        adata_perturbation: ad.AnnData,
+        sample_rep,
+        split_covariates,
+        perturbation_covariates,
+        perturbation_covariate_reps,
+    ):
+        control_key = "control"
+        sample_covariates = ["cell_type"]
+        sample_covariate_reps = {"cell_type": "cell_type"}
+        primary_group, _ = next(iter(perturbation_covariates.items()))
+        dm_old = DataManager(
+            adata_perturbation.copy(),
+            sample_rep=sample_rep,
+            split_covariates=split_covariates.copy(),
+            control_key=control_key,
+            perturbation_covariates=perturbation_covariates.copy(),
+            perturbation_covariate_reps=perturbation_covariate_reps.copy(),
+            sample_covariates=sample_covariates.copy(),
+            sample_covariate_reps=sample_covariate_reps.copy(),
+            primary_group=primary_group,
+        )
+        dm_new = DataManager(
+            adata_perturbation.copy(),
+            sample_rep=sample_rep,
+            split_covariates=split_covariates.copy(),
+            control_key=control_key,
+            perturbation_covariates=perturbation_covariates.copy(),
+            perturbation_covariate_reps=perturbation_covariate_reps.copy(),
+            sample_covariates=sample_covariates.copy(),
+            sample_covariate_reps=sample_covariate_reps.copy(),
+            primary_group=primary_group,
+        )
+
+        old = _get_condition_data_old(
+            dm=dm_old,
+            split_cov_combs=dm_old._get_split_cov_combs(adata_perturbation.obs.copy()),
+            adata=adata_perturbation.copy(),
+        )
+        new = dm_new._get_condition_data(
+            adata=adata_perturbation.copy(),
+        )
+        compare_train_data(old, new)
```

Comment:
this is to compare whether it gives the same as it used to be?

Reply:
Yes I want to be sure that the results are exactly the same for a while. Because the code isn't very interpretable unfortunately